ml-agents/mlagents/trainers/subprocess_env_manager.py

```diff
@@ -306,10 +315,31 @@ def training_behaviors(self) -> Dict[BehaviorName, BehaviorSpec]:
 
     def close(self) -> None:
         logger.debug("SubprocessEnvManager closing.")
+        for env_worker in self.env_workers:
+            env_worker.request_close()
+        # Pull messages out of the queue until every worker has CLOSED or we time out.
+        deadline = datetime.now() + WORKER_SHUTDOWN_TIMEOUT
```

Comment:
Generally I prefer `time.time()` (which returns epoch time in seconds) but that's a matter of taste.

Reply:
Switched